### PR TITLE
Feature/vr 4582 batch delete commits

### PR DESCRIPTION
**Pull request by evgeniilezhnin**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java [hunk 1]
```diff
@@ -234,6 +234,111 @@ public CommitEntity getCommitEntity(
...
+              + BranchEntity.class.getSimpleName()
+              + " br where br.id.repository_id = :repositoryId "
+              + " AND br.commit_hash IN (:commitHashes) ";
+      Query<BranchEntity> getBranchByCommitQuery =
+          session.createQuery(getBranchByCommitHQLBuilder, BranchEntity.class);
+      getBranchByCommitQuery.setParameter("repositoryId", repositoryEntity.getId());
+      getBranchByCommitQuery.setParameter("commitHashes", commitShas);
+      List<BranchEntity> branchEntities = getBranchByCommitQuery.list();
+      if (branchEntities != null && !branchEntities.isEmpty()) {
+        StringBuilder errorMessage = new StringBuilder("Commit is associated with branch name : ");
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]:
```suggestion
        StringBuilder errorMessage = new StringBuilder("Commits are associated with branch name : ");
```

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java [hunk 2]
```diff
@@ -234,6 +234,111 @@ public CommitEntity getCommitEntity(
...
+          errorMessage.append(branchEntity.getId().getBranch());
+          if (count < branchEntities.size() - 1) {
+            errorMessage.append(", ");
+          }
+          count++;
+        }
+        throw new ModelDBException(errorMessage.toString(), Code.FAILED_PRECONDITION);
+      }
+
+      String getLabelsHql =
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: deleting commit with label should be fine, when deleting the commit we need to delete the associated label mapping.
if there is some prior discussion I am missing do let me know

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java [hunk 3]
```diff
@@ -234,6 +234,111 @@ public CommitEntity getCommitEntity(
...
+      List<TagsEntity> tagsEntities = getTagsQuery.list();
+      if (tagsEntities.size() > 0) {
+        throw new ModelDBException(
+            "Commit is associated with Tags : "
+                + tagsEntities.stream()
+                    .map(tagsEntity -> tagsEntity.getId().getTag())
+                    .collect(Collectors.joining(",")),
+            Code.FAILED_PRECONDITION);
+      }
+
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java:305]: can we begin the transaction here?

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java [hunk 4]
```diff
@@ -320,8 +425,6 @@ public CommitEntity getCommitEntity(
       }
 
       session.beginTransaction();
-      // delete associated branch
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java:323]: context of this change?

**Review by ravishetye — COMMENTED** [review]
added comments

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java (hunk 4 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java:323]: there is a check before this code that a commit shouldn't be associated with a branch. So it means that a branch deletion is not needed. cc: @ad-47

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java (hunk 2 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: fixed

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java (hunk 3 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java:305]: yes, moved it here

**Review by ravishetye — APPROVED** [review]

**Pull request merged**